app/models/supporter.rb

```diff
@@ -143,22 +202,35 @@ def filled_address_fields?
   end
 
   def update_primary_address
-    if self.changes.slice(*ADDRESS_FIELDS).any? #changed an address field
```

Comment:
`self.changes.slice(*ADDRESS_FIELDS).any?` does not work here anymore because the changes are pointed to the `primary_address`.